### PR TITLE
Agent 898: Adding jobs of agent day2 install using oc binary

### PR DESCRIPTION
**Pull request by zniu1011**
1, Adding 4.18 jobs of agent day2 install using oc binary on amd64 and arm64
2, Make sure using very latest oc client
3, Update iso file name with architecture
4, Fix the arch to use the day2 architecture
5, Fix cpuArchitecture field and move it to upper layer

### CODE REVIEW

**Comment by zniu1011** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-arm64-nightly-baremetal-compact-agent-ipv4-static-fips-day2-f14

**Comment by zniu1011** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-arm64-nightly-baremetal-compact-agent-ipv4-static-fips-day2-f14

**Comment by zniu1011** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-arm64-nightly-baremetal-compact-agent-ipv4-static-fips-day2-f14

**Comment by zniu1011** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-arm64-nightly-baremetal-compact-agent-ipv4-static-fips-day2-f14

File: ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh [hunk 1]
```diff
@@ -99,16 +104,16 @@ cp "${SHARED_DIR}/nodes-config.yaml" "${DAY2_INSTALL_DIR}/"
 cp "${SHARED_DIR}/nodes-config.yaml" "${ARTIFACT_DIR}/"
 
 echo "Create node.iso for day2 worker nodes..."
-/cli/oc adm node-image create --dir="${DAY2_INSTALL_DIR}" -a "${day2_pull_secret}" --insecure=true
+/tmp/oc adm node-image create --dir="${DAY2_INSTALL_DIR}" -a "${day2_pull_secret}" --insecure=true
```
Comment (pamoedom) [inline ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh:107]: Hi @zniu1011, is the `--insecure=true` really needed here? could be possible to make it work with HTTPS or do we have some impediment?

Comment (zniu1011) [reply ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh:107]: Hi @pamoedom , it is needed here, which is used for disconnected case, to pass the self-signed certificate for local mirror registry. I didn't know other solutions here for now in this case.

Comment (pamoedom) [reply ci-operator/step-registry/baremetal/lab/agent/day2/baremetal-lab-agent-day2-commands.sh:107]: got it, it's a disconnected case, thks for the confirmation @zniu1011

**Pull request merged**